src/libsync/httplogger.cpp

```diff
@@ -146,19 +146,24 @@ void HttpLogger::logRequest(QNetworkReply *reply, QNetworkAccessManager::Operati
 
     auto ctx = std::make_unique<HttpContext>(reply->request());
 
+    const auto logError = [reply, operation, id = ctx->id]() {
+        auto url = reply->request().url();
+        qCInfo(lcNetworkHttp).noquote().nospace() << "An error occurred for " << url.toDisplayString() << ": " << reply->errorString() << " (" << reply->error()
+                                                  << ", " << operation << "), request-id: " << id;
+    };
+
     // device should still exist, lets still use a qpointer to ensure we have valid data
-    const auto logSend = [ctx = ctx.get(), operation, reply, device = QPointer<QIODevice>(device), deviceRaw = device](bool cached = false) {
+    const auto logSend = [ctx = ctx.get(), operation, reply, device = QPointer<QIODevice>(device), deviceRaw = device, logError](bool cached = false) {
         Q_ASSERT(!deviceRaw || device);
         if (!ctx->send) {
             ctx->send = true;
             ctx->timer.reset();
-        } else {
+        } else if (ctx->lastUrl != reply->url()) {
             // this is a redirect
-            if (ctx->lastUrl != reply->url()) {
-                ctx->addRedirect(reply->url());
-            } else {
-                Q_UNREACHABLE();
```

Comment:
good to see this one go - I was hitting this once ... having a client to terminate because of some (not met) condition in a logging method is just wrong.